examples/GPIO/Blink/Blink.ino

```diff
@@ -0,0 +1,40 @@
+/*  Example of reading and writing USB Host Shield GPI output
+    This example uses "Wiring" style interface. See Blink_LowLevel for example of using low-level UHS interface
+    Author: Brian Walton (brian@riban.co.uk)
+*/
+#include <UHS2_gpio.h>
+
+// Satisfy the IDE, which needs to see the include statment in the ino too.
+#ifdef dobogusinclude
+#include <spi4teensy3.h>
+#endif
+
+#define OUTPUT_PIN 0
+
+USB Usb; //Create an UHS2 interface object
+UHS2_GPIO GPIO(&Usb); //Create a GPIO object
+
+void setup()
+{
+  Serial.begin( 115200 );
+#if !defined(__MIPSEL__)
+  while (!Serial); // Wait for serial port to connect - used on Leonardo, Teensy and other boards with built-in USB CDC serial connection
+#endif
+  Serial.println("Start");
+
+  if (Usb.Init() == -1)
+    Serial.println("OSC did not start.");
+
+  delay( 200 );
+}
+
+void loop()
+{
+  //Get the current output value, toggle then wait half a second
+  int nValue = GPIO.digitalReadOutput(OUTPUT_PIN);
```

Comment:
Please use ```uint8_t``` instead of ```int```.

Reply:
The function returns int (as per the "Wiring" digitalRead). I agree this is wrong and would personally have implemented this as returning bool but I tried to be consistent with the Arduino way, which is what this PR is about. The example uses the same data type as the digitalReadOutput method which is int. The example should be consistent with the library.

Reply:
I agree, this is OK.

Reply:
One thing, indents are 8 spaces for a very specific set of reasons.
First, simply makes code easier for a human to read, which is seriously important.
Second, looks non-cluttered, and consistent across the entire library.
Finally, 8 space tabs have been a real actual standard since forever. 

Just because a few people think different, does not make it correct, especially for the above reasons.
If your text editor can accept better indents, and perform code formatting, please use those features.
Also, try to ensure that you don't have any trailing whitespace, for all the obvious reasons.

Reply:
The great thing about standards is that there are so many to choose from. Coding styles are generally chosen by a project (or, so often not chosen which is a pain) based on the lead's personal preference. I tend to use Allman (ANSI) coding style for C / C++ projects and use the Code::Blocks IDE source formatter plugin to support this style. So I generally don't cuddle braces and use 4 spaces indents which I find sufficiently clear without pushing long code lines off the screen / onto next line. When writing Java I will use the Java style and when contributing to other projects will use whatever style is either mandated / suggested or inferred from existing code (as I did for this PR). The examples in this project are a little inconsistent so may benefit from rationalising to meet your coding style. I am not aware of 8 spaces being a "real actual standard" and believe that coding style is implemented by guidelines. Please point me at any actual published standards as I have been involved in coding style guidelines for many years and would be interested to see where standards have been set. (It is many years (but not 1970's) since I learned to code and it is easy for an old man to miss the new big thing!) Cheers!

Reply:
I see your point with int, so let's stick with that.

I personally are not a big fan of 8 spaces, but that was decided a long time ago, so now we have to stick with it. And no there is no such thing as a coding style standard for C/C++ like they have in Python.

And yes some of the old examples are a little inconsistent.